torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -139,6 +140,11 @@ std::vector<at::Device> getDeviceList(const std::vector<at::Tensor>& tensors) {
   return res;
 }
 
+// Return CUDA device with ordinal given by input rank.
+at::Device getDeviceForRank(int rank) {
+  return at::Device(at::DeviceType::CUDA, rank);
```

Comment:
This probably breaks if rank is something like 13, don't you need rank % numGPUs similar to what barrier does?

Reply:
Yes, thanks for catching that!